src/algebra/lie/basic.lean

```diff
@@ -42,119 +41,99 @@ lie bracket, ring commutator, jacobi identity, lie ring, lie algebra
 
 universes u v w w₁
 
-/-- A binary operation, intended use in Lie algebras and similar structures. -/
-class has_bracket (L : Type v) := (bracket : L → L → L)
+/-- The has_bracket class has two intended uses:
+
+  1. for the product `⁅x, y⁆` of two elements `x`, `y` in a Lie algebra
+     (analogous to `has_mul` in the associative setting),
+
+  2. for the action `⁅x, m⁆` of an element `x` of a Lie algebra on an element `m` in one of its
+     modules (analogous to `has_scalar` in the associative setting). -/
```

Comment:
It strikes me as slightly odd to have one notation for the lie setting but two notations for the associative setting - for instance, should the signature of `has_mul` be changed to `A → B → B` so that it can subsume `has_scalar`?

Reply:
I too would prefer a one-to-one correspondence but I do not think we should try to subsume `has_scalar` into `has_mul`. I don't have great reasons though, really just:

- killing `has_scalar` would have the disadvantage of needing extra type ascriptions in various places,
- though (I hope!) I made it look easy, it was actually fairly tricky to set things up in such a way that we never end up with two non-defeq `has_bracket` instances, and I bet it would be a hard refactor to achieve this in the much larger library of associative algebra.